src/Yoti/Util/Profile/AnchorConverter.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Yoti\Util\Profile;
+
+use Traversable;
+use phpseclib\File\ASN1;
+use phpseclib\File\X509;
+use Attrpubapi_v1\Anchor;
+use Yoti\Entity\Anchor as YotiAnchor;
+
+class AnchorConverter
+{
+    public function __construct()
+    {
+        $this->ASN1 = new ASN1();
+        $this->X509 = new X509();
+    }
+
+    /**
+     * Convert Protobuf Anchors to a map of Yoti Anchors
+     * And merge the result into the reference array
+     *
+     * @param Anchor $anchor
+     * @param $yotiAnchorsMap
+     */
+    public function convertToYotiAnchors(Anchor $anchor, &$yotiAnchorsMap)
+    {
+        $anchorSubType = $anchor->getSubType();
+        $certificateList = $anchor->getOriginServerCerts();
+        $yotiSignedTimeStamp = $this->convertToYotiSignedTimestamp($anchor);
+        $X509CertsList = $this->convertCertsListToX509($anchor->getOriginServerCerts());
+
+        foreach ($certificateList as $certificate) {
+            $certObj = $this->convertCertToX509($certificate);
+            $certExtArr = $certObj->tbsCertificate->extensions;
+
+            if (count($certExtArr) > 1) {
+                $anchorType = $certExtArr[1]->extnId;
+                $extEncodedValue = $certExtArr[1]->extnValue;
+
+                if ($decodedAnchorValue = $this->decodedAnchorValue($extEncodedValue)) {
+                    $yotiAnchorsMap[$anchorType][] = new YotiAnchor(
+                        $decodedAnchorValue,
+                        $anchorType,
+                        $anchorSubType,
+                        $yotiSignedTimeStamp,
+                        $X509CertsList
+                    );
+                }
+            }
+        }
+    }
+
+    /**
+     * @param string $extEncodedValue
+     *
+     * @return null
+     */
+    private function decodedAnchorValue($extEncodedValue)
```

Comment:
Small thing - _decodeAnchorValue_ was correct.  Method names are better as verbs

Reply:
Good spot - That slips out of my mind.